### PR TITLE
Cleanup WAN code samples

### PR DESCRIPTION
**Pull request by mmedenjak**
- remove remnants of 3.x
- minor warning and code style fixes

### CODE REVIEW

File: enterprise/wan-replication/interactive-sample/src/main/resources/hazelcast.xml [hunk 1]
```diff
@@ -21,24 +21,11 @@
...
-                <property name="key">value</property>
-            </properties>
-        </custom-publisher>
-        <consumer>
-            <class-name>YOUR_CONSUMER_CLASS_NAME</class-name>
-            <properties>
-                <property name="key">value</property>
-            </properties>
-        </consumer>
-        -->
```
Comment (blazember) [inline enterprise/wan-replication/interactive-sample/src/main/resources/hazelcast.xml]: The readme of this sample references these configs. We should either remove that as well or keep this. I think it's good that we mention this in the sample, so I'd keep it.

Comment (mmedenjak) [reply enterprise/wan-replication/interactive-sample/src/main/resources/hazelcast.xml]: Hm, ok, I'll bring them back although I'd prefer if we had a separate sample for custom replication, this way it was a bit puzzling to me.

Comment (blazember) [reply enterprise/wan-replication/interactive-sample/src/main/resources/hazelcast.xml]: I agree that having a separate custom sample is better. My point is to keep the config consistent with the readme. I'm happy to add a custom replication sample next week and cleaning this up.

Comment (mmedenjak) [reply enterprise/wan-replication/interactive-sample/src/main/resources/hazelcast.xml]: Reverted in a separate commit.

**Comment by mmedenjak** [comment]
Thanks for the review!

**Pull request merged**